src/main/haskell/language-kore/src/Data/Kore/IndexedModule/MetadataTools.hs

```diff
@@ -0,0 +1,81 @@
+{-# LANGUAGE GADTs #-}
+{-|
+Module      : Data.Kore.IndexedModule.MetadataTools
+Description : Datastructures and functionality for retrieving metadata
+              information from patterns
+Copyright   : (c) Runtime Verification, 2018
+License     : UIUC/NCSA
+Maintainer  : traian.serbanuta@runtimeverification.com
+Stability   : experimental
+Portability : portable
+-}
+module Data.Kore.IndexedModule.MetadataTools
+    ( MetadataTools (..)
+    , extractMetadataTools
+    )
+  where
+
+import           Data.Kore.AST.Common
+import           Data.Kore.AST.Kore                    (UnifiedPattern)
+import           Data.Kore.AST.MetaOrObject
+import           Data.Kore.ASTHelpers
+import           Data.Kore.Error
+import           Data.Kore.IndexedModule.IndexedModule
+import qualified Data.Map                              as Map (Map, lookup)
+
+-- |'MetadataTools' defines a dictionary of functions which can be used to
+-- access the metadata needed during the unification process.
+data MetadataTools level = MetadataTools
+    { isConstructor    :: SymbolOrAlias level -> Bool
+    , isFunctional     :: SymbolOrAlias level -> Bool
+    , getArgumentSorts :: SymbolOrAlias level -> [Sort level]
+    , getResultSort    :: SymbolOrAlias level -> Sort level
+    }
+
+-- |'extractMetadataTools' extracts a set of 'MetadataTools' from a
+-- 'KoreIndexedModule'.  The metadata tools are functions yielding information
+-- about application heads, such as whether it is functional / constructor,
+-- or its argument and result sorts.
+--
+-- Since there is no clear consensus on how constructor and functionality
+-- axioms would be specified yet, we currently assume all symbols
+-- are constructors and all of them are functional, too.
+extractMetadataTools
+    :: MetaOrObject level
+    => KoreIndexedModule
+    -> MetadataTools level
+extractMetadataTools m = MetadataTools
+    { isConstructor = const True
+    , isFunctional = const True
+    , getArgumentSorts = applicationSortsOperands . getHeadApplicationSorts m
+    , getResultSort = applicationSortsResult . getHeadApplicationSorts m
+    }
+
+symbolSentencesMap
```

Comment:
Could you move this function, getHeadApplicationSorts and resolveThing into a new file called something like "Accesors"? Maybe also add "resolveSymbol"?